google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStubSettings.java

```diff
@@ -515,6 +563,29 @@ private Builder() {
                               .build())
                       .build());
 
+      bulkReadRowsSettings =
+          BigtableBulkReadRowsCallSettings.newBuilder(new ReadRowsBatchingDescriptor())
+              .setRetryableCodes(readRowSettings.getRetryableCodes())
+              .setRetrySettings(
+                  readRowSettings
+                      .getRetrySettings()
+                      .toBuilder()
+                      .setMaxRpcTimeout(Duration.ofSeconds(300))
+                      .setTotalTimeout(Duration.ofMinutes(20))
+                      .build())
+              .setBatchingSettings(
+                  BatchingSettings.newBuilder()
+                      .setIsEnabled(true)
+                      .setElementCountThreshold(100L)
+                      .setRequestByteThreshold(2L * 1024L * 1024)
+                      .setDelayThreshold(Duration.ofSeconds(1))
+                      .setFlowControlSettings(
+                          FlowControlSettings.newBuilder()
+                              .setLimitExceededBehavior(LimitExceededBehavior.Block)
+                              .setMaxOutstandingElementCount(1_000L)
```

Comment:
with 100 elements/batch this means that we can't have more than 10 outstanding streams. Maybe this should be 10 * 100 * numChannels? So that there can be 10 outstanding requests per channel?

Reply:
> Maybe this should be 10 * 100 * numChannels?

Sounds good except doubt for one scenario: What if enduser has say 32 channels (n1-stnd-8 machine), as each batch is unaryCallable(unlike readRows streaming) and we can not apply flow control based on actual row size. 
_Do you think it should be okay with these cases?_

We need numChannels which come from CPU so Instead of numChannels, Would `20 * 100 * numOfCPUs` be ok?

Reply:
I think we should chat tomorrow. I don't quite understand what you mean here.

Minor correction: the default settings are 2x the cpus, so -8 will have 16 channels. I think making it based on channels is more semantic